assignment4.Rmd

```diff
@@ -105,12 +111,41 @@ sweep(loadings, 2, colSums(loadings), "/") #sweep() computes each row as a propo
 
 #Now examine your components and try to come up with substantive descriptions of what some might represent?
 
+#PC1 explains the largest proportion of variance on two variables: mean_hint(30%) and mean_attempt(25%); For PC2, it accounts for the largest proportion of variance on prior_percent_correct(45%); PC3 contributes its higher share of var into mean_confidence(46%); For PC4, prior_prob_count has the highest proportion of variance (32%); For PC5, mean_attempt(36%) and problems_attempted(30%) take up bigger proportions of var; and PC6 contributes the most variance into mean_hint(36%).
+
 #You can generate a biplot to help you, though these can be a bit confusing. They plot the transformed data by the first two components. Therefore, the axes represent the direction of maximum variance. Then mapped onto this point cloud are the original directions of the variables, depicted as red arrows. It is supposed to provide a visualization of which variables "go together". Variables that possibly represent the same underlying construct point in the same direction.  
 
 biplot(pca)
 
 #Calculate values for each student that represent these your composite variables and then create a new correlogram showing their relationship to mean_correct.
 
+##not sure what it asks for
```

Comment:
This part is asking you to compose a new variable that represents the different PC's that you were describing above. We an use the <- mutate_all(as.data.frame(loadings), funs((.)/sum(.))) line from above to get the weights.

Then using those values to weigh the variable:

new <- DF$prior_prob_count*0.12423113 + DF$prior_percent_correct*0.08035956 + DF$problems_attempted*0.21744737 + DF$mean_hint*0.30223780 + DF$mean_attempt*0.25863458 + DF$mean_confidence*0.01708956